### DISCUSSION TITLE
Custom routing on server (API)

### DISCUSSION THREAD

**Discussion by TokugawaTakeshi** [nuxt/nuxt#16073] [Questions]
Thank you for the directory based routing functionality on server side (**server** directory). However, I am sorry, but I don't need it and please don't waste your time to overpersuade me.

My question is can I define the custom routing (for example [as in Express framework](https://expressjs.com/en/guide/routing.html#route-paths)) for the server part in Nuxt 3?

You may answer: "You should not use the Nuxt 3 server engine and provide the separate API application if you don't need the directory based routing".  Well, it looks like it will be such as. But is directory based routing is all functionality that Nuxt 3 server engine suggesting? According the documentation, NO. If so, IMHO the directory based routing should be the optional functionality, not required.

**Comment by ayugioh2003** [comment]
You can try this

- https://github.com/nuxt/framework/discussions/9079
- https://nuxt.com/docs/api/configuration/nuxt-config/#serverhandlers

**Reply by francisashley** [reply]
The link to the discussion is not working 😭

**Reply by TheAlexLichter** [reply]
@francisashley I think it was https://github.com/nuxt/nuxt/discussions/16365 - in contact with the GH support to fix the discussion redirects.

About the actual topic: I highly recommend to use nitro here 😋

**Reply by TokugawaTakeshi** [reply]
> About the actual topic: I highly recommend to use nitro here 😋

The Nitro is great, but I am expecting the _optional_ directory based routing from Nitro.
I will not use the directory based routing, and that's it.

**Reply by TheAlexLichter** [reply]
> > About the actual topic: I highly recommend to use nitro here 😋
> 
> The Nitro is great, but I am expecting the _optional_ directory based routing from Nitro. I will not use the directory based routing, and that's it.

You don't have to and still can use Nitro 😋
